common/components/AddDomainForm/index.tsx

```diff
@@ -0,0 +1,134 @@
+import { useJsApiLoader } from '@react-google-maps/api'
+import { Form, Input, FormInstance } from 'antd'
+import { useCallback, useState, useMemo, useRef, useEffect } from 'react'
+import { centerTownGeoCode } from 'utils/constants'
+import Map from '../Map'
+import { Polygon } from '@react-google-maps/api'
+import { IAddress } from '../../modules/models/Task'
+
+interface Props {
+  isFormDisabled: boolean
+  form: FormInstance
+}
+
+const AddDomainModal: React.FC<Props> = ({ isFormDisabled, form }) => {
+  const [waypoints, setWaypoints] = useState([])
+  const [address, setAddress] = useState<IAddress>(null)
+
+  const [libraries] = useState(['places'] as any)
+
+  const { isLoaded } = useJsApiLoader({
+    id: 'google-map-script',
+    googleMapsApiKey: process.env.NEXT_PUBLIC_GOOGLE_MAPS_API_KEY,
+    libraries,
+  })
+
+  const mapOptions = useMemo(() => {
+    return {
+      geoCode: centerTownGeoCode,
+      zoom: 17,
```

Comment:
as I understand it is an initial point, so It would be better to change zoom-property to "12".